runtime/src/main/cpp/Memory.cpp

```diff
@@ -555,6 +556,15 @@ inline void unlock(KInt* spinlock) {
 
 } // namespace
 
+ObjHeader* KRefSharedHolder::ref() const {
+  verifyRefOwner();
+#if USE_GC
+  if (obj_ != nullptr)
+    rememberNewContainer(obj_->container());
```

Comment:
There are usages of `KRefSharedHolder::ref()` that don't require `rememberNewContainer`.
Most of them now becomes suboptimal, but few of the usages are now incorrect (e.g. `KotlinBase._tryRetain`).

So we might have two variants of this function, e.g. `ref` and `localRef`, the second one is not supposed to outlive the holder and thus could be used when the reference doesn't escape.

Reply:
Seems that this change makes more harm then good. Let's think over if this behavior is indeed important.